File: geeksforgeeks/Algorithms/Searching and Sorting/selection_sort.java
import java.util.*;
public class Main {
	static void selectionSort(int arr[] , int n) //Selection Sort Function
	{
		int i,j,min_idx;
		
		for(i=0;i<n-1;i++)
		{
			min_idx=i;
			for (j = i+1; j < n; j++)
			{
				if (arr[j] < arr[min_idx])
				{
					min_idx = j;
				}
			}
			int temp=arr[min_idx];
			arr[min_idx]=arr[i];
			arr[i]=temp;
		}
	}
	
	public static void main(String[] args) 
	{
		int arr[] = {64, 25, 12, 22, 11};
		int n = arr.length;
		selectionSort(arr, n);
		
		for(int i=0;i<n;i++)
		{
			System.out.println(arr[i]);
		}
	}  
}

File: geeksforgeeks/Top 10 Algorithms and Data Structures/Searching and Sorting/Selection Sort.cpp
#include <iostream>
using namespace std;

// Selection Sort Function
void selection_sort(int arr[],int n)
{
	int min, temp;
	for(int i = 0;i < n; i++)
	{
		min = i;
		for(int j = i; j < n; j++)
		{
			if(arr[j] < arr[min])
			{
				min = j;
			}
		}
		temp = arr[min];
		arr[min] = arr[i];
		arr[i] = temp;
	}
}

int main() {
    int arr[] = {7,5,2,4,1,6,112,43,42,32121,32,1,43,2}; //Test Array
    int n = sizeof(arr)/sizeof(arr[0]);
    selection_sort(arr,n);
    
    // Sorted Output
    for(int i = 0;i < n;i++)
    {
    	cout << arr[i] << " ";
    }
    return 0;
}